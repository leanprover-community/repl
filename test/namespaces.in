{"cmd": "namespace Lean\nopen Nat\nopen Elab", "namespaces": true}

{"cmd": "open Lean\nsection test\nopen Nat\nend test", "namespaces": true, "env": 0}
