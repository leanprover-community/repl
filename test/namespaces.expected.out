{"namespaces":
 [{"pp": "namespace Lean",
   "pos": {"line": 1, "column": 0},
   "openDecls": [],
   "endPos": {"line": 1, "column": 14},
   "currentNamespace": "Lean"},
  {"pp": "open Nat",
   "pos": {"line": 2, "column": 0},
   "openDecls": ["Nat"],
   "endPos": {"line": 2, "column": 8},
   "currentNamespace": "Lean"},
  {"pp": "open Elab",
   "pos": {"line": 3, "column": 0},
   "openDecls": ["Lean.Elab", "Nat"],
   "endPos": {"line": 3, "column": 9},
   "currentNamespace": "Lean"}],
 "env": 0}

{"namespaces":
 [{"pp": "open Lean",
   "pos": {"line": 1, "column": 0},
   "openDecls": ["Lean", "Lean.Elab", "Nat"],
   "endPos": {"line": 1, "column": 9},
   "currentNamespace": "Lean"},
  {"pp": "section test",
   "pos": {"line": 2, "column": 0},
   "openDecls": ["Lean", "Lean.Elab", "Nat"],
   "endPos": {"line": 2, "column": 12},
   "currentNamespace": "Lean"},
  {"pp": "open Nat",
   "pos": {"line": 3, "column": 0},
   "openDecls": ["Nat", "Lean", "Lean.Elab", "Nat"],
   "endPos": {"line": 3, "column": 8},
   "currentNamespace": "Lean"},
  {"pp": "end test",
   "pos": {"line": 4, "column": 0},
   "openDecls": ["Lean", "Lean.Elab", "Nat"],
   "endPos": {"line": 4, "column": 8},
   "currentNamespace": "Lean"}],
 "env": 1}

